hyperspy/_signals/electron_wave_image.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2016 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import numpy as np
+
+from hyperspy._signals.complex_signal import ComplexSignal
+
+
+class ElectronWaveImage(ComplexSignal):
+    """Signal2D subclass for complex electron wave data (e.g. reconstructed from holograms)."""
+
+    _record_by = "image"
+    _signal_type = "electron wave"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.axes_manager.set_signal_dimension(2)

```

Comment:
If we go for a `ComplexSignal2D` subclass, it'll be there were this line should go.